components/cosmetic_filters/resources/data/content_cosmetic.ts

```diff
@@ -553,45 +559,15 @@ const scheduleQueuePump = (hide1pContent: boolean, generichide: boolean) => {
   }, { timeout: maxTimeMSBeforeStart })
 }
 
-const vettedSearchEngines = ['duckduckgo', 'qwant', 'bing', 'startpage', 'google', 'yandex', 'ecosia']
-
-chrome.runtime.onMessage.addListener((msg, sender, sendResponse) => {
-  const action = typeof msg === 'string' ? msg : msg.type
-  switch (action) {
-    case 'cosmeticFilteringBackgroundReady': {
-      if (msg.hideOptions !== undefined) {
-        scheduleQueuePump(msg.hideOptions.hide1pContent, msg.hideOptions.generichide)
-      }
-      injectScriptlet(msg.scriptlet)
-      break
-    }
-    case 'cosmeticFilterConsiderNewSelectors': {
-      const { selectors } = msg
-      let nextIndex = cosmeticStyleSheet.rules.length
-      for (const selector of selectors) {
-        if (_parsedCurrentDomain.type === ParseResultType.Listed && vettedSearchEngines.includes(_parsedCurrentDomain.icann.domain)) {
-          continue
-        }
-        if (allSelectorsToRules.has(selector)) {
-          continue
-        }
-        // insertRule always adds to index 0,
-        // so we always add to end of list manually.
-        cosmeticStyleSheet.insertRule(
-          `${selector}{display:none !important;}`,
-          nextIndex
-        )
-        allSelectorsToRules.set(selector, nextIndex)
-        nextIndex++
-        firstRunQueue.add(selector)
-      }
-      // @ts-ignore
-      if (!document.adoptedStyleSheets.includes(cosmeticStyleSheet)) {
-        // @ts-ignore
-        document.adoptedStyleSheets = [cosmeticStyleSheet]
-      }
-      scheduleQueuePump(false, false)
-      break
-    }
-  }
-})
+if (!window.content_cosmetic.observingHasStarted) {
+  window.content_cosmetic.observingHasStarted = true
+  scheduleQueuePump(window.content_cosmetic.hide1pContent,
+    window.content_cosmetic.generichide)
+} else if (window.content_cosmetic.scriplet &&
+    window.content_cosmetic.scriplet !== '') {
```

Comment:
Redundant check: empty string is falsey

Reply:
that check is needed as the script is executed several times, so we don't reapply scriptlets every time

Reply:
Understood, but my point was that if `window.content_cosmetic.scriplet == true` then `window.content_cosmetic.scriplet !== ''` is constant (and impossible for scriplet to still be `''` since `'' == false`).

Reply:
ok, my bad